src/modules/appeals-service/AppealsService.ts

```diff
@@ -37,6 +42,38 @@ export class AppealsService {
 
     }
 
+    public async hasExistingAppeal(companyNumber: string, penaltyReference: string, accessToken: string,
+                                   refreshToken: string): Promise<boolean> {
+
+        const operation = 'get';
+
+        this.checkArgumentOrThrow(companyNumber, 'Company number is missing');
+        this.checkArgumentOrThrow(penaltyReference, 'penalty reference is missing');
+        this.checkArgumentOrThrow(accessToken, 'Access token is missing');
+        this.checkArgumentOrThrow(refreshToken, 'Refresh token is missing');
+
+        this.refreshTokenInterceptor(accessToken, refreshToken);
+
+        const uri: string = `${this.uri}/companies/${companyNumber}/appeals?`;
```

Comment:
is the '?' needed? Or you could continue to build this url with the penalty reference? - then your debug statement would be more helpful

Reply:
I dont think it is necessary, ill update the debug statement tho